robot-name/robot-name_test.spec.js

```diff
@@ -12,9 +12,17 @@ describe("Robot", function() {
   });
 
   it("different robots have different names", function() {
-    var robotOne = new Robot();
-    var robotTwo = new Robot();
-    expect(robotOne.name).not.toEqual(robotTwo.name);
+    var usedNames= {};
+
+    var i = 0,
+        max = 10000;
+
+    for (i; i < max; i++) {
```

Comment:
Yeah, JSLint rules are sometimes controversial. And you'll find people on both sides of the discussion. I'd write the `for` loop the following way, that is in between the two edge cases we're talking here, declare the variable at the top, initialize it in the loop:

```javascript
it("different robots have different names", function() {
  var i,
    max = 10000,
    usedNames= {};

  for (i = 0; i < max; i++) {
    // ...
  }
}
```